src/redux/ensRegistration.ts

```diff
@@ -0,0 +1,266 @@
+) => async (dispatch: AppDispatch, getState: AppGetState) => {
+  const {
+    ensRegistration: { registrations, currentRegistrationName },
+  } = getState();
+  const lcAccountAddress = accountAddress.toLowerCase();
+
+  const accountRegistrations = registrations?.[lcAccountAddress] || {};
+  const registration = accountRegistrations[currentRegistrationName] || {};
+  const registrationRecords = registration?.records || {};
+
+  const updatedEnsRegistrationManagerForAccount = {
+    registrations: {
+      ...registrations,
+      [lcAccountAddress]: {
+        ...accountRegistrations,
+        [currentRegistrationName]: {
+          ...registration,
+          records: { ...registrationRecords, [key]: value },
+        },
+      },
+    },
+  };
+  dispatch({
+    payload: updatedEnsRegistrationManagerForAccount,
+    type: ENS_REGISTRATION_UPDATE_RECORDS,
+  });
+};
+
+export const removeRecordByKey = (
+  accountAddress: EthereumAddress,
+  key: string
+) => async (dispatch: AppDispatch, getState: AppGetState) => {
+  const {
+    ensRegistration: { registrations, currentRegistrationName },
+  } = getState();
+
+  const lcAccountAddress = accountAddress.toLowerCase();
+  const accountRegistrations = registrations?.[lcAccountAddress] || {};
+  const registration = accountRegistrations[currentRegistrationName] || {};
+  const registrationRecords = registration?.records || {};
+
+  const newRecords = omit(registrationRecords, key) as Records;
+
+  const updatedEnsRegistrationManagerForAccount = {
+    registrations: {
+      ...registrations,
+      [lcAccountAddress]: {
+        ...accountRegistrations,
+        [currentRegistrationName]: {
+          ...registration,
+          records: newRecords,
+        },
+      },
+    },
+  };
+
+  dispatch({
+    payload: updatedEnsRegistrationManagerForAccount,
+    type: ENS_REGISTRATION_REMOVE_RECORD_BY_KEY,
+  });
+};
+
+export const saveCommitRegistrationParameters = (
+  accountAddress: EthereumAddress,
+  registrationParameters: RegistrationParameters
+) => async (dispatch: AppDispatch, getState: AppGetState) => {
+  const {
+    ensRegistration: { registrations, currentRegistrationName },
+  } = getState();
+
+  const lcAccountAddress = accountAddress.toLowerCase();
+  const accountRegistrations = registrations?.[lcAccountAddress] || {};
+  const registration = accountRegistrations[currentRegistrationName] || {};
+  const updatedEnsRegistrationManagerForAccount = {
+    registrations: {
+      ...registrations,
+      [lcAccountAddress]: {
+        ...accountRegistrations,
+        [currentRegistrationName]: {
+          ...registration,
+          ...registrationParameters,
+        },
+      },
+    },
+  };
+
+  dispatch({
+    payload: updatedEnsRegistrationManagerForAccount,
+    type: ENS_SAVE_COMMIT_REGISTRATION_PARAMETERS,
+  });
+};
+
+// -- Reducer ----------------------------------------- //
+const INITIAL_STATE: ENSRegistrationState = {
+  currentRegistrationName: '',
```

Comment:
note to self: want to double check no paths for race condition on which should be the "current registration name" when a user triggers multiple registrations per or across wallets

Reply:
this is meant just while in the registration flow to be able to access the registration data from state, this is why there's only one string here, since the user can only be in one registration flow at a time, even tho multiple registrations can be happening in the background. 
Here, `startRegistration` is taking the current wallet address to access that piece of data in a future iteration when  we're persisting the registration state you would be able to `continueRegistration` and the flow would take the user to the corresponding registration state depending on the state of the registration